### PR TITLE
Clientmgr mini refactorr

### PR DESCRIPTION
**Pull request by christf**

### CODE REVIEW

**Comment by neocturne** [comment]
Generally, I'm not a fan of many of the renamings. Names should be as long as necessary to be clearly understandable (for someone who knows what a function is supposed to do), but not longer.

Some names are excessively long; some changes just make types and variables longer, but not easier to understand (e.g. ip_address instead of ip). I also liked the old "plen", as it's a very common abbreviation of "prefix length".

I'll also add a few comments inline.

File: src/clientmgr.c [hunk 1]
```diff
@@ -13,37 +13,33 @@
...
+	return ipv6_address;
 }
 
 bool prefix_contains(const struct prefix *prefix, struct in6_addr *addr) {
-	int plen = prefix->plen;
 
-	for (int i = 0; i < 16; i++) {
-		int mask = ~((1<<(8 - (plen > 8 ? 8 : plen))) - 1);
+	for (int remaining_prefix_length = prefix->length, i=0;remaining_prefix_length > 0; remaining_prefix_length-= 8) {
+		int mask = ~((1<<(8 - (remaining_prefix_length > 8 ? 8 : remaining_prefix_length))) - 1);
```
Comment (neocturne) [inline src/clientmgr.c]: This line is quite ugly, we should find a way to clean it up.

File: src/clientmgr.c [hunk 2]
```diff
@@ -67,21 +63,21 @@ void print_client(struct client *client) {
 		printf(")\n");
 	}
 
-	for (int i = 0; i < VECTOR_LEN(client->addresses); i++) {
-		struct client_ip *e = &VECTOR_INDEX(client->addresses, i);
+	for (int i = 0; i < VECTOR_LEN(client->ip_addresses); i++) {
+		struct client_ip_address *client_ip_address_iterator = &VECTOR_INDEX(client->ip_addresses, i);
```
Comment (neocturne) [inline src/clientmgr.c]: *client_ip_address_iterator* is much too long, and it is not an iterator. I think *e* is fine here, as it's short for "element"; *addr* and *ip* would also be options.

File: src/clientmgr.c [hunk 3]
```diff
@@ -13,37 +13,33 @@
...
+
+	return ipv6_address;
 }
 
 bool prefix_contains(const struct prefix *prefix, struct in6_addr *addr) {
-	int plen = prefix->plen;
 
-	for (int i = 0; i < 16; i++) {
-		int mask = ~((1<<(8 - (plen > 8 ? 8 : plen))) - 1);
+	for (int remaining_prefix_length = prefix->length, i=0;remaining_prefix_length > 0; remaining_prefix_length-= 8) {
```
Comment (neocturne) [inline src/clientmgr.c]: Please move the logic changes to a separate commit.

File: src/clientmgr.h [hunk 4]
```diff
@@ -7,36 +7,40 @@
...
-	int plen;
+	int length; /* in bits */
 };
 
-struct client_ip {
-	enum ip_state state;
-	int tentative_cnt;
+struct client_ip_address {
+	enum ipaddress_state state;
+	int tentative_remaining_retries;
```
Comment (neocturne) [inline src/clientmgr.h]: A bit long, maybe "tentative_rem_retries" or "tentative_retries_left"?

**Review by neocturne — CHANGES_REQUESTED** [review]

File: src/clientmgr.c (hunk 3 above)
Comment (christf) [reply src/clientmgr.c]: will do. when re-formatting I will keep a whole copy of the new function in there as prefix_contains_v2 just to make refactoring and renaming things easier but it will be omitted in the prettify-commit itself.

File: src/clientmgr.c (hunk 1 above)
Comment (christf) [reply src/clientmgr.c]: done.

File: src/clientmgr.h (hunk 4 above)
Comment (christf) [reply src/clientmgr.h]: done

File: src/clientmgr.c (hunk 2 above)
Comment (christf) [reply src/clientmgr.c]: done.

**Comment by christf** [comment]
@NeoRaider Danke für das Feedback, die Punkte sind umgesetzt.

**Comment by neocturne** [comment]
If you want another review, please squash your fixups.

**Comment by christf** [comment]
commits are squashed now. Another review would be nice.
